src/model.jl

```diff
@@ -801,29 +647,25 @@ Add an `:initial`, `:final`, `:control`, `:state` or `:variable` box constraint
 
 ```jldoctest
 julia> constraint!(ocp, :initial, rg=2:3, lb=[ 0, 0 ], ub=[ 1, 2 ])
-julia> constraint!(ocp, :final, val=Index(1), lb=0, ub=2)
-julia> constraint!(ocp, :control, val=Index(1), lb=0, ub=2)
+julia> constraint!(ocp, :final, rg=1, lb=0, ub=2)
+julia> constraint!(ocp, :control, rg=1, lb=0, ub=2)
 julia> constraint!(ocp, :state, rg=2:3, lb=[ 0, 0 ], ub=[ 1, 2 ])
 julia> constraint!(ocp, :initial, rg=1:2:5, lb=[ 0, 0, 0 ], ub=[ 1, 2, 1 ])
 julia> constraint!(ocp, :variable, rg=1:2, lb=[ 0, 0 ], ub=[ 1, 2 ])
 ```
 """
 function constraint!(ocp::OptimalControlModel{<: TimeDependence, <: VariableDependence}, type::Symbol; 
-    rg::Union{RangeConstraint,Nothing}=nothing, f::Union{Function,Nothing}=nothing,
-    val::Union{ctVector,Nothing}=nothing, lb::Union{ctVector,Nothing}=nothing, ub::Union{ctVector,Nothing}=nothing, 
+    rg::Union{OrdinalRange{<:Integer},Integer,Nothing}=nothing, f::Union{Function,Nothing}=nothing, lb::Union{ctVector,Nothing}=nothing, ub::Union{ctVector,Nothing}=nothing, 
     label::Symbol=__constraint_label())
 
-    
     (lb ≢ nothing && ub === nothing) && (ub = Inf*(size(lb,1) == 1 ? 1 : ones(eltype(ub), size(ub,1))))
     (lb === nothing && ub ≢ nothing) && (lb = -Inf*(size(ub,1) == 1 ? 1 : ones(eltype(ub), size(ub,1))))
-
-    @match (rg,f,val,lb,ub) begin
-        (::Nothing,::Nothing,::ctVector,::Nothing,::Nothing) => return constraint!(ocp, type, val, label) #
-        (::Nothing,::Nothing,::Nothing,::ctVector,::ctVector) => return constraint!(ocp, type, lb, ub, label) #
-        (::Nothing,::Function,::ctVector,::Nothing,::Nothing) => return constraint!(ocp, type, f, val, label) #
-        (::Nothing,::Function,::Nothing,::ctVector,::ctVector) => return constraint!(ocp, type, f, lb, ub, label) #
-        (::RangeConstraint,::Nothing,::ctVector,::Nothing,::Nothing) => return constraint!(ocp, type, rg, val, label) #
-        (::RangeConstraint,::Nothing,::Nothing,::ctVector,::ctVector) => return constraint!(ocp, type, rg, lb, ub, label) #
+    (typeof(rg) <: Int) && (rg = Index(rg))
+    
+    @match (rg,f,lb,ub) begin
+        (::Nothing,::Nothing,::ctVector,::ctVector) => return constraint!(ocp, type, lb, ub, label) #
```

Comment:
@BaptisteCbl at this point, I see that you reduce the computation of the new `constraint!` functions with updated named arguments to three cases/subfunctions:
- linear, full range (no range give implies full range)
- linear, given range
- nonlinear (function)

Wouldn't it be more robust, particularly with less **code duplication** (*e.g.*, check on arguments), to do the converse:
- fully code the new version with named args
- implement the other calls of `constraint!` (namely the three cases mentioned above) by calling this unique new function?